### PR TITLE
fix : fixes #669 by updating main_window.py

### LINKED ISSUE #669: Can't cancel during saving file.
### Information

- **Qiskit Metal version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
Choose `save as` but instead of saving a file, click cancel. The app will crash.

### Steps to reproduce the problem
Choose `save as` but instead of saving a file, click cancel. The app will crash.

### What is the expected behavior?
Choose `save as` but instead of saving a file, click cancel. The app should close the file-name-choosing window and not save a copy of the chip

### PR DESCRIPTION
**Pull request by chauhan-shobhit**
What are the issues this pull addresses (issue numbers / links)?

Issue #669
#669

Did you add tests to cover your changes (yes/no)?

no

Did you update the documentation accordingly (yes/no)?

yes

Did you read the CONTRIBUTING document (yes/no)?

yes

Summary

Fixed the issue #669 by updating save_design and save_design_copy function in qiskit_metal/_gui/main_window.py

Details and comments

When using cancel instead of save, QFileDialog.getSaveFileName returns empty string( https://github.com/Qiskit/qiskit-metal/blob/main/qiskit_metal/_gui/main_window.py#L163 ) as filename.
This caused a FileNotFoundError during opening of file (https://github.com/Qiskit/qiskit-metal/blob/main/qiskit_metal/_gui/main_window.py#L171 and https://github.com/Qiskit/qiskit-metal/blob/main/qiskit_metal/_gui/main_window.py#L148 )

To fix the issue , an empty check was added. File will be opened and written to only if filename is non-empty (length of string is non-zero)

### CODE REVIEW

**Comment by chauhan-shobhit** [comment]
Did any of the reviewers get a chance to look at this ? 
Any feedback ?

**Comment by priti-ashvin-shah-ibm** [comment]
@chauhan-shobhit  Thank you for your time contribute to Qiskit Metal.  The CI check has failed and there appears to be a formatting error.  We use yapf to format the code so it "looks" similar for all contributors. I think if you format your contribution to yapf, it will pass CI check. 

 The screen shot may help.  The image came from the link at:
https://github.com/Qiskit/qiskit-metal/blob/main/CONTRIBUTING.md  

image

**Comment by chauhan-shobhit** [comment]
@priti-ashvin-shah-ibm 
My bad. I should have taken care of this earlier. 
Thanks for your time and feedback. I have updated the file applying yapf checks. Can you please review and trigger the pipeline ?

**Comment by priti-ashvin-shah-ibm** [comment]
> 
Are you a first time contributor?  Did you sign the CLA -Contributor License Agreement?

https://github.com/Qiskit/qiskit-metal/blob/main/CONTRIBUTING.md

**Comment by chauhan-shobhit** [comment]
@priti-ashvin-shah-ibm 
Yes. 
I am a first time contributor and I have signed the CLA.

**Comment by priti-ashvin-shah-ibm** [comment]
> @priti-ashvin-shah-ibm Yes. I am a first time contributor and I have signed the CLA.

Usually, when someone signs it for the first time, I see it in the pull request.  That is why I asked if you had signed it.

**Comment by chauhan-shobhit** [comment]
> > @priti-ashvin-shah-ibm Yes. I am a first time contributor and I have signed the CLA.
> 
> Usually, when someone signs it for the first time, I see it in the pull request. That is why I asked if you had signed it.

I remember signing it when I first submitted the pull request. 
I tried https://cla-assistant.io/Qiskit/qiskit (from contributing page) but it doesn't have any options for submitting response probably because it's already signed. Version field is empty and there is no dropdown. Screenshot below for reference.

If there any other place where I need to sign it, please share ?

image

**Comment by priti-ashvin-shah-ibm** [comment]
@chauhan-shobhit Please check the email address you used for signing the CLA.

- Could be that  you don't have an email in your Qiskit Metal github profile
- Could be that you signed with a different email that is in used in github.

**Comment by priti-ashvin-shah-ibm** [comment]
@chauhan-shobhit  There was an issue with the CLA.  Can you please sign it again with the email address used in the personal profile you used to access github for the code edits?

**Comment by mtreinish** [comment]
@chauhan-shobhit @priti-ashvin-shah-ibm the lack of a CLA assistant comment here was not related to having previously signed the cla or not. It was an infrastructure issue with the cla assistant service. We don't run the bot and just use the service and this happens semi-regularly. However, in an attempt to manually re-trigger the bot I noticed a configuration issue with cla assistant on this repository which prevented me from doing so. In the process of fixing that issue the cla assistant database entries got swapped with the shared db entry for all of qiskit and the entries previously used on the repository. So now some contributors will need to resign the cla for the bot to be happy now, sorry for the inconvenience.

**Comment by priti-ashvin-shah-ibm** [comment]
> @chauhan-shobhit @priti-ashvin-shah-ibm the lack of a CLA assistant comment here was not related to having previously signed the cla or not. It was an infrastructure issue with the cla assistant service. We don't run the bot and just use the service and this happens semi-regularly. However, in an attempt to manually re-trigger the bot I noticed a configuration issue with cla assistant on this repository which prevented me from doing so. In the process of fixing that issue the cla assistant database entries got swapped with the shared db entry for all of qiskit and the entries previously used on the repository. So now some contributors will need to resign the cla for the bot to be happy now, sorry for the inconvenience.

Thank You @mtreinish for your help and support.

**Comment by chauhan-shobhit** [comment]
@priti-ashvin-shah-ibm @mtreinish 
I have signed the CLA. Can you please check again ?
Thanks for your patience.


image

File: qiskit_metal/_gui/main_window.py [hunk 1]
```diff
@@ -145,8 +145,10 @@ def save_design_copy(self):
 
         # save python script to file path
         pyscript = self.design.to_python_script()
-        with open(filename, 'w') as f:
-            f.write(pyscript)
+        #check whether filename is empty or not. Save file only when filename is non-empty.
+        if len(filename):
```
Comment (zlatko-minev) [inline qiskit_metal/_gui/main_window.py:149]: Probably if filename is enough

Comment (zlatko-minev) [reply qiskit_metal/_gui/main_window.py:149]: good to check indeed!

**Review by zlatko-minev — APPROVED** [review]
Nice!

**Pull request merged**